### ISSUE TITLE
Invalid type error while trying to parse process.env object

### ISSUE THREAD

**Issue by vladshcherbin** [fabian-hiller/valibot#587]
Code:

```ts
import process from 'node:process'
import { object, parse, string } from 'valibot'

console.info(typeof process.env)
console.info(process.env.NODE)
console.info('---')

parse(
  object({
    NODE: string()
  }),
  process.env
)
```

Result:

<img width="1301" alt="image" src="https://github.com/fabian-hiller/valibot/assets/6711845/b00955c7-64f8-4517-932e-4af3e164a027">

- same error with `looseObject` and `strictObject`
- no error in 0.30
- no error with spread object copy:

```ts
parse(
  object({
    NODE: string()
  }),
  { ...process.env }
)
```

valibot - 0.31.0-rc.0
node - 20.10.0

**Comment by vladshcherbin** [comment]
Cause: [Last check from object](https://github.com/fabian-hiller/valibot/blob/v0.31.0-rc.0/library/src/schemas/object/object.ts#L100) doesn't pass.

**Comment by fabian-hiller** [comment]
I made the check stricter to only match plain objects. I don't know why `process.env` is not a plain object. `Object.getPrototypeOf(process.env).constructor.name` returns an empty string. Not sure if we should make the check less strict because of this.

**Comment by vladshcherbin** [comment]
I also have no idea and it'll be strange if it doesn't pass.

Out of interest quick test:
- `zod` - pass
- https://www.npmjs.com/package/is-plain-object - doesn't pass
- https://www.npmjs.com/package/is-plain-obj - doesn't pass

**Comment by fabian-hiller** [comment]
If others encounter the same problem, I would like to hear their opinions and whether the workaround is sufficient.

**Comment by fabian-hiller** [comment]
From #587. Happy to hear your feedback.

> There are other problems with the current plain object check, for example in #587, but I am not sure if Valibot is the problem. Normally, this should not be a problem. We could remove the `input.constructor === Object` check to make it less strict, but then any object that is not `null` will be accepted. In practice, this is usually not a problem because we never return the original object. So, strictly speaking, the validation is still safe. Therefore, I might make the validation less strict so as not to have to deal with special JS runtime behavior that we can't control.

**Comment by fabian-hiller** [comment]
This is fixed in v0.31.0-rc.6

**Comment by vladshcherbin** [comment]
Perfect, thank you!

I think the fix is good until a valid non-false object assumption is found 👍
